### ISSUE TITLE
Import-Module with UseWindowsPowerShell parameter as LOCALSYSTEM account fails if $HOME\Documents\Powershell\Modules directory does not exist.

### ISSUE THREAD

**Issue by cmichaelingram** [PowerShell/PowerShell#17744]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

We have a Windows Service that runs as LOCALSYSTEM that uses the PowerShell SDK to periodically run pwsh commands.  The calls to import modules with the UseWindowsPowershell parameter are failing. 

I was able to narrow the down the scope of the repro steps to what I have below.  I used the ActiveDirectory module as my example to import, but we are also seeing this behavior on other modules we import with the UseWindowsPowershell parameter.  

1. Open pwsh as LOCAL SYSTEM Account.  (I used `psexec -s -i pwsh.exe` from a command prompt on a test machine).
2. Run `Import-Module ActiveDirectory -UseWindowsPowerShell`.  If you don't have the ActiveDirectory module installed, see here: https://docs.microsoft.com/en-us/powershell/module/activedirectory/?view=windowsserver2022-ps.  I've also found running this command will produce the same results,` Import-Module Microsoft.Powershell.Management -UseWindowsPowerShell -NoClobber`
3. Import-Module fails.
![image](https://user-images.githubusercontent.com/31104435/180496333-af95b9f6-c5de-486e-bffb-97d162cc2b57.png)
4. Message logged to the PowerShell Core Operational Event logs is misleading as it seems to indicate it failed to create module in c:\windows\temp folder.
![image](https://user-images.githubusercontent.com/31104435/180497039-023f0912-3376-4512-b1fb-df17adfd9b9e.png)

WORKAROUND:
1. Close the pwsh window.  
2. Create the directory: c:\windows\system32\config\systemprofile\documents\powershell\modules.  C:\windows\system32\config\systemprofile is the $HOME folder for the LOCAL SYSTEM account.  
3. Open pwsh as LOCAL SYSTEM Account (I used `psexec -s -i pwsh.exe `from a command prompt).
4. Run `Import-Module ActiveDirectory -UseWindowsPowerShell `
5. ActiveDirectory module successfully imports.
![image](https://user-images.githubusercontent.com/31104435/180496613-288a8c0e-985f-4bed-8c67-007221d8ac22.png)

### Expected behavior

```console
Module is successfully imported.
```

### Actual behavior

```console
Module is not imported and a failure message is returned.
```

### Error details

```console
Import-Module: Failed to generate proxies for remote module 'ActiveDirectory'. The -OutputModule parameter does not resolve to a path, and a user module path cannot be found for the provided name.
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.2.5
PSEdition                      Core
GitCommitId                    7.2.5
OS                             Microsoft Windows 10.0.17763
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by daxian-dbw** [comment]
I think this was fixed by #17671. The PS7.2.6+ versions have the fix.

**Comment by cmichaelingram** [comment]
> I think this was fixed by #17671. The PS7.2.6+ versions have the fix. Please try out the latest 7.2.11. Feel free to open a new issue if you still run into the same problem with 7.2.11.

@daxian-dbw I tested just now with 7.2.11 and can confirm that this is resolved.  Thank you!
